### PR TITLE
Line info for errors

### PR DESCRIPTION
**Pull request by mterwoord**
Better approach to include line numbers in errors (I think).

Exposes an issue with XamlReader. Included a test case for it (but probably needs moving around).

### CODE REVIEW

File: src/Portable.Xaml/Portable.Xaml/XamlObjectWriter.cs [hunk 1]
```diff
@@ -235,6 +233,13 @@ public override void WriteStartMember (XamlMember property)
 				return;
 			}
 
+			if (property.IsUnknown)
+				throw new XamlObjectWriterException($"Cannot set unknown member '{property}'")
+				      {
+					      LineNumber = line,
+						  LinePosition = column
+				      };
+
```
Comment (cwensley) [inline src/Portable.Xaml/Portable.Xaml/XamlObjectWriter.cs]: Is there a test for this case?

**Comment by cwensley** [comment]
Thanks for submitting the PR!  The tests should go in the appropriate test class, such as XamlObjectWriterTest.cs as it the PropertyNotFound tests are testing the functionality of the XamlObjectWriter. Some questions that I have are:

- Are we certain that `ShouldProvideLineInfo` always returns true? 
- Does the XamlServices.Transform set the line info in System.Xaml?  It'd be good to verify this is the right place to set the line info

Some tests would be great to verify the above as well.

**Comment by mterwoord** [comment]
Regarding setting the line info: It could be moved to ```XamlWriter.WriteNode```, but then we have to cast each time.

Same goes for ```ShouldProvideLineInfo```: If we move it, we have to do checks each node, and that costs a lost of performance. Besides that, the ```XamlServices.Transform``` method is the one doing the loop over the reader, so it feels like a correct place to do things..

According to ```ShouldProvideLineInfo```, according to MSDN, XamlObjectWriter should always have it return true. See [MSDN](https://msdn.microsoft.com/en-us/library/system.xaml.xamlobjectwriter.shouldprovidelineinfo(v=vs.110).aspx)

I'm updating the PR to move the tests

**Comment by cm4ker** [comment]
This PR superseeded by #123.

**Pull request closed without merging**